### PR TITLE
domain,metrics: move domain metrics, add privilege load counter

### PR DESCRIPTION
**Pull request by tiancaiamao**
Move the `metrics.go` file and add a LoadPrivilegeCounter metric.

### CODE REVIEW

**Comment by shenli** [comment]
LGTM

File: metrics/domain.go [hunk 1]
```diff
@@ -11,32 +11,44 @@
...
 			Help:      "Bucketed histogram of processing time (s) in load schema.",
 			Buckets:   prometheus.ExponentialBuckets(0.001, 2, 15),
 		})
+
+	// LoadPrivilegeCounter records the counter of load privilege.
+	LoadPrivilegeCounter = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Namespace: "tidb",
+			Subsystem: "domain",
+			Name:      "load_privilege_total",
```
Comment (zimulala) [inline metrics/domain.go]: s/load_privilege_total/load_privilege ?

Comment (tiancaiamao) [reply metrics/domain.go]: It's a counter, so `_total` is OK, you can see similar name rules in their best practice:
https://prometheus.io/docs/practices/naming/

**Pull request merged**